server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -432,8 +433,8 @@ public MappedFieldType fieldType(String fullName) {
     /**
      * Returns all mapped field types.
      */
-    public Iterable<MappedFieldType> fieldTypes() {
-        return this.mapper == null ? Collections.emptySet() : this.mapper.mappers().fieldTypes();
+    public Iterable<MappedFieldType> fieldTypes(Predicate<MappedFieldType> predicate) {
+        return this.mapper == null ? Collections.emptySet() : this.mapper.mappers().fieldTypes().filter(predicate);
     }
```

Comment:
Given that we *only* use this for index warmers, maybe we should replace this with a `getEagerGlobalOrdinalsFields()` method that returns a set of field names, and which calls FieldTypeLookup.filter() under the hood?

Reply:
yes I agree. will do that